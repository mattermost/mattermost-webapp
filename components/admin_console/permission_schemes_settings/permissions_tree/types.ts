// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type Permission = {
    id: string;
    combined?: boolean;
    permissions: string[];
}
export type Group = {
    id: string;
    permissions: Array<Permission | string>;
}

export type AdditionalValues = {
    edit_post: {
        editTimeLimitButton: JSX.Element;
    };
}
