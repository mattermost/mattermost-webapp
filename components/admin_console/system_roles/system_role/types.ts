// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type ReadAccess = 'read';
export const readAccess: ReadAccess = 'read';
export type WriteAccess = 'write';
export const writeAccess: WriteAccess = 'write';
export type NoAccess = false;
export const noAccess: NoAccess = false;
export type MixedAccess = 'mixed';
export const mixedAccess: MixedAccess = 'mixed';

export type PermissionAccess = ReadAccess | WriteAccess | NoAccess;
export type PermissionsToUpdate = Record<string, ReadAccess | WriteAccess | NoAccess>;
export type PermissionToUpdate = {
    name: string;
    value: PermissionAccess;
};

export type SystemSection = {
    name: string;
    hasDescription?: boolean;
    subsections?: SystemSection[];
    disabled?: boolean;
}
