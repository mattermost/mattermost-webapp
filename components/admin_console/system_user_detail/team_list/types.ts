// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Team, TeamMembership} from '@mattermost/types/teams';

export type TeamWithMembership = Team & TeamMembership;
