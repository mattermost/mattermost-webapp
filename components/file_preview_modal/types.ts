// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {FileInfo} from '@mattermost/types/files';

export type LinkInfo = {
    has_preview_image: boolean;
    link: string;
    extension: string;
    name: string;
}

export function isFileInfo(info: FileInfo | LinkInfo): info is FileInfo {
    return Boolean((info as FileInfo).id);
}
