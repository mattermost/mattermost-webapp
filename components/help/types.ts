// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export enum HelpLink {
    Messaging = 'Messaging',
    Composing = 'Composing',
    Mentioning = 'Mentioning',
    Formatting = 'Formatting',
    Attaching = 'Attaching',
    Commands = 'Commands',
}
