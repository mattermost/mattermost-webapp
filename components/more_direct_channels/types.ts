// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Channel} from '@mattermost/types/channels';
import {UserProfile} from '@mattermost/types/users';

import {Value} from 'components/multiselect/multiselect';

// Not to be confused with the GroupChannel type used for LDAP groups
export type GroupChannel = Channel & {
    profiles: UserProfile[];
}

export function isGroupChannel(option: UserProfile | GroupChannel): option is GroupChannel {
    return (option as GroupChannel)?.type === 'G';
}

export type Option = (UserProfile & {last_post_at?: number}) | GroupChannel;

export type OptionValue = Option & Value;

export function optionValue(option: Option): OptionValue {
    return {
        value: option.id,
        label: isGroupChannel(option) ? option.display_name : option.username,
        ...option,
    };
}
