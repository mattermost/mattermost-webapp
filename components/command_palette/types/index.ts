// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export enum CommandPaletteEntities {
    Channel = 'channel',
    Playbooks = 'playbooks',
    Boards = 'boards',
    Messages = 'messages',
    Files = 'files',
    Runs = 'runs',
    GoTo = 'goTo',
}

export enum ChannelType {
    DM_CHANNEL= 'D',
    GM_CHANNEL= 'G',
    OPEN_CHANNEL= 'O',
    PRIVATE_CHANNEL= 'P',
}

export enum BoardsType {
    OPEN_CHANNEL= 'O',
    PRIVATE_CHANNEL= 'P',
}

export enum PlaybooksType {
    OPEN_CHANNEL= 'O',
    PRIVATE_CHANNEL= 'P',
}

