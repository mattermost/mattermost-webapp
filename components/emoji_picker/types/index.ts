// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {EmojiCategory, Emoji} from 'mattermost-redux/types/emojis';

export type Category = {
    className: string;
    emojiIds?: string[];
    id: string;
    message: string;
    name: EmojiCategory;
    offset: number;
};

export type Categories = Record<EmojiCategory, Category>;

export type CategoryOrEmojiRow = {
    type: 'categoryHeaderRow' | 'emojisRow';
    row: EmojiCategory | Emoji[];
}
