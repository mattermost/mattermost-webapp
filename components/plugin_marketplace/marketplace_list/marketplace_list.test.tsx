// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {AuthorType, MarketplacePlugin, ReleaseStage} from 'mattermost-redux/types/marketplace';

import MarketplaceItem from '../marketplace_item/marketplace_item_plugin';

import MarketplaceList from './marketplace_list';
import NavigationRow from './navigation_row';

describe('components/marketplace/marketplace_list', () => {
    const samplePlugin: MarketplacePlugin = {
        homepage_url: 'https://github.com/mattermost/mattermost-plugin-nps',
        download_url: 'https://github.com/mattermost/mattermost-plugin-nps/releases/download/v1.0.3/com.mattermost.nps-1.0.3.tar.gz',
        author_type: AuthorType.Mattermost,
        release_stage: ReleaseStage.Production,
        enterprise: false,
        manifest: {
            id: 'com.mattermost.nps',
            name: 'User Satisfaction Surveys',
            description: 'This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost',
            version: '1.0.3',
            min_server_version: '5.14.0',
        },
        installed_version: '',
    };

    it('should render with multiple plugins', () => {
        const wrapper = shallow<MarketplaceList>(
            <MarketplaceList
                listing={[
                    samplePlugin, samplePlugin, samplePlugin, samplePlugin, samplePlugin,
                    samplePlugin, samplePlugin, samplePlugin, samplePlugin, samplePlugin,
                    samplePlugin, samplePlugin, samplePlugin, samplePlugin, samplePlugin,
                    samplePlugin, samplePlugin,
                ]}
            />,
        );
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.state().page).toEqual(0);
        expect(wrapper.find(MarketplaceItem)).toHaveLength(15);
        expect(wrapper.find(NavigationRow)).toHaveLength(1);
        expect(wrapper.find(NavigationRow).props().page).toEqual(0);
        expect(wrapper.find(NavigationRow).props().total).toEqual(17);
        expect(wrapper.find(NavigationRow).props().maximumPerPage).toEqual(15);
    });

    it('should set page to 0 when list of plugins changed', () => {
        const wrapper = shallow<MarketplaceList>(
            <MarketplaceList
                listing={[samplePlugin, samplePlugin]}
            />,
        );

        wrapper.setState({page: 10});
        wrapper.setProps({listing: [samplePlugin]});

        expect(wrapper.state().page).toEqual(0);
    });
});
