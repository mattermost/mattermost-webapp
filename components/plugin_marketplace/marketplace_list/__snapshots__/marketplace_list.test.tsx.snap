// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/marketplace/marketplace_list should render with multiple plugins 1`] = `
<div
  className="more-modal__list"
>
  <Connect(MarketplaceItemPlugin)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItemPlugin)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItemPlugin)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItemPlugin)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItemPlugin)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItemPlugin)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItemPlugin)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItemPlugin)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItemPlugin)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItemPlugin)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItemPlugin)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItemPlugin)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItemPlugin)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItemPlugin)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItemPlugin)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(NavigationRow)
    maximumPerPage={15}
    onNextPageButtonClick={[Function]}
    onPreviousPageButtonClick={[Function]}
    page={0}
    total={17}
  />
</div>
`;
